### PR TITLE
Reporting: Update docs

### PR DESCRIPTION
**Pull request by sakjur**
+ Recommending sending a test email after report is saved for verification.
+ Added details about the different scheduling options.
+ Documented the experimental API we encourage users to use.
+ Image scale factor.

Feel free to apply grammar and style fixes directly on the PR.

Out-of-scope for this particular PR:
- Update documentation to mention that reporting is an early version still under heavy development. (#24045)
- Add examples to make it clear what to expect of a report. (#24045)

### CODE REVIEW

File: docs/sources/enterprise/reporting.md
Comment (jongyllen) [inline docs/sources/enterprise/reporting.md:116]:
```suggestion
When generating reports, each panel renders separately before being collected in a PDF. The per panel rendering timeout and number of concurrently rendered panels can be configured.
```

File: docs/sources/enterprise/reporting.md
Comment (jongyllen) [inline docs/sources/enterprise/reporting.md:42]: I would like to add that it also can be used to verify that emails actually can be sent. Not sure how to phrase it though

Comment (sakjur) [reply docs/sources/enterprise/reporting.md:42]:
```suggestion
1. **Send test mail** after saving the report to verify that the report works like you expect it to.
```

mayhaps?

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md:116]:
```suggestion
When generating reports each panel renders separately before being collected in a PDF. The per panel rendering timeout and number of concurrently rendered panels can be configured.
```

Do we need that comma?

**Review by jongyllen — APPROVED** [review]
Looks good. Maybe comma, maybe not. We can probably live with this

**Comment by sakjur** [comment]
:+1: Yeah, I guess.

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [reply docs/sources/enterprise/reporting.md:42]:
```suggestion
1. **Send test mail** after saving the report to verify that the whole configuration is working as expected.
```
Maybe? I agree that otherwise it's a bit redundant with the previous step `Preview the report to make sure it appears as you expect. Update if necessary`
And I think it should be "send test email" but this is what in the UI. Should we update it?

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:50]:
```suggestion
All scheduling indicates when the reporting service will start rendering the dashboard. It can take a few minutes to render a dashboard with a lot of panels.
```
It feels weird to start by `All scheduling is approximative`. It's not very positive about the feature.

**Comment by AgnesToulet** [comment]
Looks good!

We also discussed here: https://github.com/grafana/grafana-enterprise/pull/472#discussion_r400021280 about timerange issue. I tried to write a paragraph to explain this but not sure it's very clear. And I don't know where we should put this paragraph.

Preview:
>### Dashboard data timerange
>
>The dashboard data timerange is displayed at the top of the report. If your dashboard is set to use local browser timezone, it will use the Grafana server timezone.
>
> > If your Grafana server and your Renderer Plugin server are not in the same timezone, it could lead to two different displays between the top timerange and the graphs timeranges. In this case, we advise you to set your dashboard timezone to UTC to have display consistency.

Markdown:
```
### Dashboard data timerange

The dashboard data timerange is displayed at the top of the report. If your dashboard is set to use local browser timezone, it will use the Grafana server timezone.

> If your Grafana server and your Renderer Plugin server are not in the same timezone, it could lead to two different displays between the top timerange and the graphs timeranges. In this case, we advise you to set your dashboard timezone to UTC to have display consistency.
```

Or should we just add a paragraph in the troubleshooting section like `Different timezones in report timeranges`?

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md:50]: 🤔 I kind of prefer my variant here. For two reasons. One is that it's important to note that the _start_ of rendering might not be on the dot, it may shift back and forth a little based on a number of reasons (mostly forwards). The other thing is that one of the things we said we wanted to improve was setting clear expectations for the feature, and I think it'll lead to less frustration when the documentation acknowledges tradeoffs straight away.

**Comment by sakjur** [comment]
I think that looks pretty good, I'll try to incorporate that and see where it makes the most sense?

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [reply docs/sources/enterprise/reporting.md:50]: I prefer @AgnesToulet 's suggested edit.

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:64]:
```suggestion
* **Time zone -** Time zone for the **Time** argument.
```

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:54]: I am confused by your use of "arguments" here. Are these options in the UI that you fill in or select? If so, then they are probably fields. 

This is end user documentation, not developer notes.

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:72]:
```suggestion
* **Time zone -** Time zone for the **Time** argument.
```

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:93]:
```suggestion
## Reports API
```

Comment (oddlittlebird) [reply docs/sources/enterprise/reporting.md:93]: Are there going to be more Enterprise-only APIs? If so, then we will need to properly document this and others in a separate subsection.

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md:54]: I feel as I cannot verbatim just replace “arguments” with “fields” here, but it makes perfect sense that arguments is something that isn’t used outside of the command-line/API terminology. I’ll dig through the docs a little and write up something hopefully more user-centric tomorrow.

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md:93]: Yes, that’s all but certain. Do you think we should add that already or should we wait until we have more/stable documented APIs? (I somewhat prefer the latter, otoh maybe that would mess up URLs or something like that?)

I’ll update the title.

Comment (oddlittlebird) [reply docs/sources/enterprise/reporting.md:93]: The thing is, we will need to remember to split this API out into a separate document when we get another one. It would mess up any links to that heading as well.

Comment (sakjur) [reply docs/sources/enterprise/reporting.md:93]: Then I’ll split it up and link to the API doc from here 🙂

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md:50]: I'll merge that then 🤷 But I still don't understand why, so I _will_ most likely do the same mistake in my next PR. I've learnt nothing here, and my response was specifically spelling out my reasons for my choices to open up for learning (even if it's just "it just feels better").

Comment (sakjur) [reply docs/sources/enterprise/reporting.md]: Just talked with @xlson and explained my concerns with the difference and he pointed out that my concern about the approximative start position. He'll open an issue with more details and we'll fix it in the backend instead of documenting it.

**Review by jongyllen — APPROVED** [review]
Nice work!

File: docs/sources/enterprise/http_api.md
Comment (AgnesToulet) [inline docs/sources/enterprise/http_api.md:21]:
```suggestion
### Send report
```
I think the "manually" part is included in the fact that it's an API endpoint

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:95]:
```suggestion
Sending reports programmatically is possible with the [send report]({{< relref "http_api.md#send-report" >}}) endpoint in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```
If you approve my comment above

**Comment by AgnesToulet** [comment]
> 
> 
> I think that looks pretty good, I'll try to incorporate that and see where it makes the most sense?

I think you forgot about that? Or you didn't find anywhere to put it?

**Comment by sakjur** [comment]
Yes! I did 😊 I'll add that 👍

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:64]:
```suggestion
* **Time zone -** Time zone for the **Time** field.
```

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:72]:
```suggestion
* **Time zone -** Time zone for the **Time** field.
```

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:78]:
```suggestion
Reports which are scheduled to never be sent have no parameter and will not be sent to the scheduler. They may be manually generated from the **Send test email** prompt or via the API.
```

**Review by AgnesToulet — COMMENTED** [review]
Nice work! Last comments around arguments / fields and I think we're good to go :)

**Comment by oddlittlebird** [comment]
@AgnesToulet , "time range" is two words.

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:13]: Add an intro here and to each h2. It can be short, but it should set the context so the reader knows what to expect in the document or section.

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:65]:
```suggestion
Code | Description
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:63]:
```suggestion
### Status codes
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:53]:
```suggestion
### Example response
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:47]:
```suggestion
Field name | Type | Description
```

Comment (oddlittlebird) [reply docs/sources/enterprise/http_api.md:47]: Can you be more specific about "type"? Data type? Entry type?

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:45]:
```suggestion
### JSON body schema
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:31]:
```suggestion
### Example request
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:21]:
```suggestion
### Send a report
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:20]: Need an intro here. Something like

This API allows you to interact programmatically with the [Reporting](LINK) feature.

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:29]: You need to lead with this. Is it just reports/email that is experimental, or is it the whole reporting API? Put it as high up in the hierarchy as makes sense.

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:27]:
```suggestion
Generate and send a report. This API waits for the report to be generated before returning. We recommend that you set the client's timeout to at least 60 seconds.
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:36]:
```suggestion
Content-type: application/json
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:49]:
```suggestion
id | string | ID of the report to send. It is the same as in the URL when editing a report, not to be confused with the ID of the dashboard. Required.
```

File: docs/sources/enterprise/http_api.md
Comment (oddlittlebird) [inline docs/sources/enterprise/http_api.md:67]: Put periods at the ends of these sentences.

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:95]:
```suggestion
You can send reports programmaticallywith the [Reporting API]({{< relref "http_api.md#send-report" >}}) in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
I was unclear about the API doc. 

Like in the HTTP APIs section, we will eventually need one topic per API. This one looks good, except that it should be in a documentation section called Enterprise APIs (in the sidebar) and preferably in its own folder as well with an _index.md file. This will give us a good place to store future Enterprise API docs as well.

The current H2 Reporting API should be switched to an H1, then move the paragraph about Authentication down to under the bits about what the Reporting API does and the experimental bit.

File: docs/sources/enterprise/http_api.md
Comment (sakjur) [reply docs/sources/enterprise/http_api.md:36]: The name of the header is `Content-Type`, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

File: docs/sources/enterprise/http_api.md
Comment (sakjur) [reply docs/sources/enterprise/http_api.md]: Just the `POST /api/reports/email` endpoint.

File: docs/sources/enterprise/reporting.md
Comment (sakjur) [reply docs/sources/enterprise/reporting.md]: I'd like to refer to the exact endpoint, not the API as a whole, made the change to active voice but kept the endpoint reference.

**Comment by sakjur** [comment]
I noticed there is a precedence for putting Enterprise APIs under the HTTP APIs, so I did that with this one as well. The section about enterprise APIs still needs an introduction, but that's out of scope for this PR.

I've tried to apply or respond to all change requests that has come in so far and feel pretty confident that this is in a state where it's good enough for us to merge. I'll wait until tomorrow to gather some final feedback before merging 🙂

File: docs/sources/enterprise/reporting.md
Comment (oddlittlebird) [inline docs/sources/enterprise/reporting.md:122]:
```suggestion
If the time zone is set differently between your Grafana server and its remote image renderer, then the time ranges in the report might be different between the page header and the time axes in the panels. We advise always setting the time zone to UTC for dashboards when using a remote renderer to avoid this. 
```

Comment (oddlittlebird) [reply docs/sources/enterprise/reporting.md:122]: Use "might" instead of "may" when talking about possibilities. "May" has connotations of permission. "Yes, you may do that" vs. "That might happen."

File: docs/sources/http_api/reporting.md
Comment (oddlittlebird) [inline docs/sources/http_api/reporting.md:16]: Switch the order of these two first paragraphs.

File: docs/sources/http_api/reporting.md
Comment (oddlittlebird) [inline docs/sources/http_api/reporting.md:18]:
```suggestion
## Send a report
```

**Review by oddlittlebird — APPROVED** [review]
Looks great. I added a few minor edits.

File: docs/sources/http_api/reporting.md
Comment (sakjur) [reply docs/sources/http_api/reporting.md]: Whoops, had forgotten to push that fix 😇

File: docs/sources/enterprise/reporting.md
Comment (AgnesToulet) [inline docs/sources/enterprise/reporting.md:118]:
```suggestion
Reports use the saved time range of the dashboard. Changing the time range of the report is done by saving a modified time range to the dashboard.
```

**Review by AgnesToulet — APPROVED** [review]
LGTM! I added one last catch.

**Pull request merged**
